### PR TITLE
Plugin interface cleanup

### PR DESCRIPTION
**Pull request by bjorn3**
The first commit performs two uncontroversial cleanups. The second commit removes `#[plugin_registrar]` and instead requires you to export a `__rustc_plugin_registrar` function, this will require a change to servo's script_plugins (cc @jdm)

### CODE REVIEW

File: compiler/rustc_feature/src/removed.rs [hunk 1]
```diff
@@ -136,9 +136,12 @@ declare_features! (
     (removed, main, "1.53.0", Some(29634), None, None),
     (removed, pub_macro_rules, "1.53.0", Some(78855), None,
      Some("removed due to being incomplete, in particular it does not work across crates")),
-     /// Allows the definition of `const` functions with some advanced features.
+    /// Allows the definition of `const` functions with some advanced features.
     (removed, const_fn, "1.54.0", Some(57563), None,
      Some("split into finer-grained feature gates")),
+    /// Allows using `#[plugin_registrar]` on functions.
+    (removed, plugin_registrar, "1.54.0", Some(29597), None,
+     Some("an __rustc_plugin_registrar symbol must now be defined instead")),
```
Comment (petrochenkov) [inline compiler/rustc_feature/src/removed.rs:144]:
```suggestion
     Some("a __rustc_plugin_registrar symbol must now be defined instead")),
```
?

**Comment by petrochenkov** [comment]
I think removing `#[plugin_registrar]` is fine.
The functionality is deeply deprecated and as long as servo still can access it somehow it doesn't matter much how exactly it happens, through an attribute or through an exported symbol.

**Comment by petrochenkov** [comment]
Could you add a test making sure that two plugins from two different crates still can be loaded together despite the identical names of the plugin registrar symbols?

**Comment by crlf0710** [comment]
@bjorn3 Ping from triage, any updates on this?

**Comment by JohnCSimon** [comment]
@bjorn3 Ping again from triage, any updates on this?

**Comment by bjorn3** [comment]
Rebased, but I still need to at the requested test.

**Comment by petrochenkov** [comment]
r=me with the failing test updated and commits squashed.

**Comment by bjorn3** [comment]
Note to self: Prepare a servo PR before this gets merged.

**Comment by bjorn3** [comment]
Opened https://github.com/servo/servo/pull/28564 for the servo changes.

> r=me with the failing test updated and commits squashed.

Fixed all failing tests and squashed all commits.

@bors r=petrochenkov

**Pull request merged**